appinventor/buildserver/src/com/google/appinventor/buildserver/resources/runtime.scm

```diff
@@ -943,6 +943,41 @@
   ;;      )
   ;;  '(100 200 17 300))
 
+(define-syntax map_nondest
+  (syntax-rules ()
+    ((_ lambda-arg-name body-form list)
+     (yail-list-map (lambda (lambda-arg-name) body-form) list))))
+
+;;(define-syntax map_nondest
+;;  (syntax-rules ()
+;;    ((_ lambda-arg-name body-form list)
+;;     (yail-list-map (lambda (lambda-arg-name) body-form) list))))
+
+(define-syntax filter_nondest
+  (syntax-rules ()
+    ((_ lambda-arg-name body-form list)
+     (yail-list-filter (lambda (lambda-arg-name) body-form) list))))
+
+;;(define-syntax filter_dest
+;;  (syntax-rules ()
+;;    ((_ lambda-arg-name body-form list)
+;;     (yail-list-filter! (lambda (lambda-arg-name) body-form) list))))
```

Comment:
Was there a `yail-list-filter!` in Soojin's original branch?  If so, it would be nice to include it in the code below but comment it out.  Similarly for `yail-list-map!` which I'm guessing was used in the definition of `map_dest` (see my earlier comment about that).

Reply:
Fixed